src/luaotfload-auxiliary.lua

```diff
@@ -558,12 +620,31 @@ function aux.provides_feature(font_id,        asked_script,
     return false
   end
   asked_script    = stringlower(asked_script)
-  asked_language  = stringlower(asked_language)
   asked_feature   = lpegmatch(strip_garbage, asked_feature)
 
-  if font_id > 0 then
-    local tfmdata  = identifiers[font_id]
-    if not tfmdata then return false end
+  local tfmdata = identifiers[font_id]
+  if not tfmdata then
+    logreport ("log", 0, "aux", "no font with id %d", font_id)
+    return false
+  end
+  local hbface = get_hbface(tfmdata)
+  if hbface then
+    asked_language  = stringupper(asked_language)
+    local script = harf.Tag.new(asked_script)
+    -- HarfBuzz expects "DFLT" in lower-case
+    local language = harf.Tag.new(asked_language == "DFLT" and "dflt" or asked_language)
```

Comment:
It is the otherway around here, the language tag should be `dflt`, though in this case it is a convention since the default language does not have an actual tag and is stored in the font binary in a special record.